conversion_test.go

```diff
@@ -1216,38 +1218,68 @@ func TestEnDecode(t *testing.T) {
 	type jsonStruct struct {
 		Foo *Int
 	}
+	type jsonBigStruct struct {
+		Foo *big.Int
+	}
 	var testSample = func(i int, bigSample big.Int, intSample Int) {
 		// Encoding
 		wantHex := fmt.Sprintf("0x%s", bigSample.Text(16))
 		wantDec := bigSample.Text(10)
 
-		if got := intSample.Hex(); wantHex != got {
-			t.Fatalf("test %d #1, got %v, exp %v", i, got, wantHex)
+		if have, want := intSample.Hex(), fmt.Sprintf("0x%s", bigSample.Text(16)); have != want {
+			t.Fatalf("test %d #1, have %v, want %v", i, have, want)
 		}
-		if got := intSample.String(); wantHex != got {
-			t.Fatalf("test %d #2, got %v, exp %v", i, got, wantHex)
+		if have, want := intSample.String(), bigSample.String(); have != want {
+			t.Fatalf("test %d String(), have %v, want %v", i, have, want)
 		}
-		if got, _ := intSample.MarshalText(); wantHex != string(got) {
-			t.Fatalf("test %d #3, got %v, exp %v", i, got, wantHex)
+		{
+			have, _ := intSample.MarshalText()
+			want, _ := bigSample.MarshalText()
+			if !bytes.Equal(have, want) {
+				t.Fatalf("test %d MarshalText, have %q, want %q", i, have, want)
+			}
 		}
-		if got, _ := intSample.Value(); wantDec != got.(string) {
-			t.Fatalf("test %d #4, got %v, exp %v", i, got, wantHex)
+		{
+			have, _ := intSample.MarshalJSON()
+			want := []byte(fmt.Sprintf(`"%s"`, bigSample.Text(10)))
+			if !bytes.Equal(have, want) {
+				t.Fatalf("test %d MarshalJSON, have %q, want %q", i, have, want)
+			}
+		}
+		if have, _ := intSample.Value(); wantDec != have.(string) {
+			t.Fatalf("test %d #4, got %v, exp %v", i, have, wantHex)
 		}
-		if got := intSample.Dec(); wantDec != got {
-			t.Fatalf("test %d #5, got %v, exp %v", i, got, wantHex)
+		if have, want := intSample.Dec(), wantDec; have != want {
+			t.Fatalf("test %d Dec(), have %v, want %v", i, have, want)
 		}
 		{ // Json
 			jsonEncoded, err := json.Marshal(&jsonStruct{&intSample})
 			if err != nil {
-				t.Fatalf("test %d #6, err: %v", i, err)
+				t.Fatalf("test %d: json encoding err: %v", i, err)
 			}
+			jsonEncodedBig, _ := json.Marshal(&jsonBigStruct{&bigSample})
+			//if !bytes.Equal(jsonEncoded, jsonEncodedBig) {
```

Comment:
Any specific reason for this big commented out chunk of code?

Reply:
not really